pkg/exporter/adminstats.go

```diff
@@ -0,0 +1,226 @@
+package exporter
+
+import (
+	"context"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/google/go-github/v32/github"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/promhippie/github_exporter/pkg/config"
+)
+
+// AdminStatsCollector collects metrics about GHE instances
+type AdminStatsCollector struct {
+	client   *github.Client
+	logger   log.Logger
+	failures *prometheus.CounterVec
+	duration *prometheus.HistogramVec
+	config   config.Target
+
+	// repos
+	ReposTotal  *prometheus.Desc
+	ReposRoot   *prometheus.Desc
+	ReposForked *prometheus.Desc
+	ReposOrg    *prometheus.Desc
+	PushedTotal *prometheus.Desc
+	WikisTotal  *prometheus.Desc
+	// hooks
+	HooksTotal    *prometheus.Desc
+	HooksActive   *prometheus.Desc
+	HooksInactive *prometheus.Desc
+	// pages
+	PagesTotal *prometheus.Desc
+	// orgs
+	OrgsTotal            *prometheus.Desc
+	OrgsDisabled         *prometheus.Desc
+	OrgsTeamsTotal       *prometheus.Desc
+	OrgsTeamMembersTotal *prometheus.Desc
+	// users
+	UsersTotal     *prometheus.Desc
+	UsersAdmin     *prometheus.Desc
+	UsersSuspended *prometheus.Desc
+	// pulls
+	PullsTotal       *prometheus.Desc
+	PullsMerged      *prometheus.Desc
+	PullsMergeable   *prometheus.Desc
+	PullsUnmergeable *prometheus.Desc
+	// issues
+	IssuesTotal  *prometheus.Desc
+	IssuesOpen   *prometheus.Desc
+	IssuesClosed *prometheus.Desc
+	// milestones
+	MilestonesTotal  *prometheus.Desc
+	MilestonesOpen   *prometheus.Desc
+	MilestonesClosed *prometheus.Desc
+	// gists
+	GistsTotal   *prometheus.Desc
+	GistsPrivate *prometheus.Desc
+	GistsPublic  *prometheus.Desc
+	// comments
+	CommentsCommitTotal      *prometheus.Desc
+	CommentsGistTotal        *prometheus.Desc
+	CommentsIssueTotal       *prometheus.Desc
+	CommentsPullRequestTotal *prometheus.Desc
+}
+
+// NewAdminStatsCollector returns a new AdminStatsCollector.
+func NewAdminStatsCollector(logger log.Logger, client *github.Client, failures *prometheus.CounterVec, duration *prometheus.HistogramVec, cfg config.Target) *AdminStatsCollector {
+	failures.WithLabelValues("org").Add(0)
```

Comment:
I noticed this should be `adminstats`. Let me know if you have any other feedback before I push a fix